src/main/java/com/jfrog/ide/common/persistency/ScanCache.java

```diff
@@ -110,10 +110,8 @@ private void addComponents(Map<String, ? extends Component> components, Severity
             }
             // If not exist, creates a new data object.
             GeneralInfo info = new GeneralInfo(id, component.getImpactPaths().get(0).get(0).getFullPath(), packageType);
-            Artifact artifact = new Artifact(info, new HashSet<>() {{
-                add(issue);
-            }}, new HashSet<>());
-            this.add(artifact);
+            Artifact artifact = new Artifact(info, Sets.newHashSet(issue), new HashSet<>());
```

Comment:
Something I noticed about the Artifact class - 
Its constructor receieve two hashsets, but inside the class, these hashets are also initialized upon defibition, whicg seem redundant. I therefore suggest changing - 
```
public class Artifact implements Serializable {
    private static final long serialVersionUID = 1L;
    private GeneralInfo generalInfo = new GeneralInfo();
    private Set<Issue> issues = new HashSet();
    private Set<License> licenses = new HashSet();

    public Artifact() {
    }

    public Artifact(GeneralInfo generalInfo, Set<Issue> issues, Set<License> licenses) {
        this.generalInfo = generalInfo;
        this.issues = issues;
        this.licenses = licenses;
    }
```
into -
```
public class Artifact implements Serializable {
    private static final long serialVersionUID = 1L;
    private GeneralInfo generalInfo;
    private Set<Issue> issues;
    private Set<License> licenses;

    public Artifact() {
       generalInfo = new GeneralInfo();
       issues = new HashSet();
       licenses = new HashSet();
    }

    public Artifact(GeneralInfo generalInfo, Set<Issue> issues, Set<License> licenses) {
        this.generalInfo = generalInfo;
        this.issues = issues;
        this.licenses = licenses;
    }
```

Reply:
https://github.com/jfrog/build-info/pull/565